internal/awsdata/lambda.go

```diff
@@ -0,0 +1,77 @@
+package awsdata
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/manywho/awsinventory/internal/inventory"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// AssetTypeLambdaFunction is the value used in the AssetType field when fetching Lambda functions
+	AssetTypeLambdaFunction string = "Lambda Function"
+
+	// ServiceLambda is the key for the Lambda service
+	ServiceLambda string = "lambda"
+)
+
+func (d *AWSData) loadLambdaFunctions(region string) {
+	defer d.wg.Done()
+
+	lambdaSvc := d.clients.GetLambdaClient(region)
+
+	log := d.log.WithFields(logrus.Fields{
+		"region":  region,
+		"service": ServiceLambda,
+	})
+
+	log.Info("loading data")
+
+	var functions []*lambda.FunctionConfiguration
+	done := false
+	params := &lambda.ListFunctionsInput{}
+	for !done {
+		out, err := lambdaSvc.ListFunctions(params)
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+
+		functions = append(functions, out.Functions...)
+
+		if out.NextMarker == nil {
+			done = true
+		} else {
+			params.Marker = out.NextMarker
+		}
+	}
+
+	log.Info("processing data")
+
+	for _, f := range functions {
+		var vpcId string
+		if f.VpcConfig != nil {
+			vpcId = aws.StringValue(f.VpcConfig.VpcId)
+		}
+
+		d.results <- result{
+			Row: inventory.Row{
+				UniqueAssetIdentifier:          aws.StringValue(f.FunctionName),
+				Virtual:                        true,
+				BaselineConfigurationName:      aws.StringValue(f.Version),
+				OSNameAndVersion:               "Amazon Linux",
+				Location:                       region,
+				AssetType:                      AssetTypeLambdaFunction,
+				HardwareMakeModel:              fmt.Sprintf("%ds, %dMB", aws.Int64Value(f.Timeout), aws.Int64Value(f.MemorySize)),
```

Comment:
Taking a bit of an artistic license to this column... trying to provide some useful information about the specific resource. Welcome thoughts...